hazelcast/src/main/java/com/hazelcast/core/Cluster.java

```diff
@@ -84,6 +84,18 @@
     Member getLocalMember();
 
     /**
+     * Promotes local lite member to data member.
+     * When this method returns both {@link #getLocalMember()} and {@link #getMembers()}
+     * reflects the promotion.
+     *
+     * @throws IllegalStateException if member is not a lite member
+     * @throws IllegalStateException if mastership claim in progress
+     * @throws IllegalArgumentException if local member cannot be identified as a member of the cluster
```

Comment:
hmm... I can change `IllegalArgumentException ` to `IllegalStateException `.  `IllegalArgumentException` is thrown by `MembershipManager#promoteToNormalMember` and that exception is delegated to the caller.